presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/cache/CachingHiveMetastore.java

```diff
@@ -324,27 +317,31 @@ private PartitionStatistics loadPartitionColumnStatistics(WithIdentity<HiveParti
                 partition.getKey().getHiveTableName().getDatabaseName(),
                 partition.getKey().getHiveTableName().getTableName(),
                 ImmutableSet.of(partitionName));
-        if (!partitionStatistics.containsKey(partitionName)) {
+        PartitionStatistics result = partitionStatistics.get(partitionName);
+        if (result == null) {
             throw new PrestoException(HIVE_PARTITION_DROPPED_DURING_QUERY, "Statistics result does not contain entry for partition: " + partition.getKey().getPartitionName());
         }
-        return partitionStatistics.get(partitionName);
+        return result;
     }
 
     private Map<WithIdentity<HivePartitionName>, PartitionStatistics> loadPartitionColumnStatistics(Iterable<? extends WithIdentity<HivePartitionName>> keys)
     {
         SetMultimap<WithIdentity<HiveTableName>, WithIdentity<HivePartitionName>> tablePartitions = stream(keys)
-                .collect(toImmutableSetMultimap(value -> new WithIdentity<>(value.getIdentity(), value.getKey().getHiveTableName()), key -> key));
+                .collect(toImmutableSetMultimap(value -> new WithIdentity<>(value.getIdentity(), value.getKey().getHiveTableName()), Function.identity()));
         ImmutableMap.Builder<WithIdentity<HivePartitionName>, PartitionStatistics> result = ImmutableMap.builder();
         tablePartitions.keySet().forEach(table -> {
-            Set<String> partitionNames = tablePartitions.get(table).stream()
+            Set<WithIdentity<HivePartitionName>> partitionNames = tablePartitions.get(table);
+            ImmutableSet<String> partitionNameStrings = partitionNames.stream()
                     .map(partitionName -> partitionName.getKey().getPartitionName().get())
                     .collect(toImmutableSet());
-            Map<String, PartitionStatistics> partitionStatistics = delegate.getPartitionStatistics(table.getIdentity(), table.getKey().getDatabaseName(), table.getKey().getTableName(), partitionNames);
-            for (String partitionName : partitionNames) {
-                if (!partitionStatistics.containsKey(partitionName)) {
-                    throw new PrestoException(HIVE_PARTITION_DROPPED_DURING_QUERY, "Statistics result does not contain entry for partition: " + partitionName);
+            Map<String, PartitionStatistics> partitionStatistics = delegate.getPartitionStatistics(table.getIdentity(), table.getKey().getDatabaseName(), table.getKey().getTableName(), partitionNameStrings);
+            for (WithIdentity<HivePartitionName> partitionName : partitionNames) {
+                String stringNameForPartition = partitionName.getKey().getPartitionName().get();
+                PartitionStatistics statsFromMap = partitionStatistics.get(stringNameForPartition);
```

Comment:
How about we rename `partitionStatistics` to `statisticsByPartitionName` and then rename `statsFromMap` to `partitionStatistics`?

Reply:
Sure thing.